scripts/scripts.js

```diff
@@ -156,6 +156,22 @@ function buildAutoBlocks(main) {
     console.error('Auto Blocking failed', error);
   }
 }
+
+/**
+ * Creates an optimized background image for the given section.
+ * The image is created for the given breakpoints.
+ * Default breakpoints are mobile, tablet and desktop.
+ * @param section The section to create the background image for
+ * @param bgImage The background image to optimize
+ * @param breakpoints The breakpoints to optimize the image for
+ */
+function createOptimizedBackgroundImage(section, bgImage, breakpoints = [{ width: '450' }, { media: '(min-width: 450px)', width: '750' }, { media: '(min-width: 750px)', width: '2000' }]) {
```

Comment:
i think the function has two issues:
- `br.width` is applied without variable name ("?750" instead of "?width=750")
- `pathname` needs to be url-encoded

this could be fixed as follows:
```
function createOptimizedBackgroundImage(section, bgImage, breakpoints = [{ width: '450' }, { media: '(min-width: 450px)', width: '750' }, { media: '(min-width: 750px)', width: '2000' }]) {
  const url = new URL(bgImage, window.location.href);
  const pathname = encodeURIComponent(url.pathname);

  const images = breakpoints.map((br, i) => `url(${pathname}?width=${br.width}&format=webply&optimize=medium) ${i + 1}x`);
  section.style.backgroundImage = `image-set(${images.join(", ")})`;
  section.style.backgroundSize = 'cover';
}
```